packages/frontend-v2/src/services/dataMapping.service.ts

```diff
@@ -0,0 +1,31 @@
+export const planTypeMapper = (id: number): string => {
+  let result: string = '';
+  switch (id) {
+    case 1:
+      result = 'hebdomadaire';
+      break;
+    case 2:
+      result = 'mensuel';
+      break;
+    case 3:
+      result = 'trimestriel';
+      break;
+    case 4:
+      result = 'semestriel';
+      break;
+    case 5:
+      result = 'annuel';
+      break;
+    default:
+      result = 'autre';
+      break;
+  }
+  return result;
+};
+export enum PlanType {
+  'hebdomadaire',
+  'mensuel',
+  'trimestriel',
+  'semestriel',
+  'annuel',
+}
```

Comment:
If we use https://www.typescriptlang.org/docs/handbook/enums.html#numeric-enums all this code can be a lot more clean. And why have you multiple time PlanType declaration in the frontend ?